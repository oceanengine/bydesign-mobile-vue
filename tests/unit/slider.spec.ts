import { mount, trigger, triggerDrag, mockGetBoundingClientRect } from '../index';
import { Slider as NativeSlider } from '@src';
const Slider = NativeSlider as any;
import Vue from 'vue';
function mockRect(vertical = null) {
  return mockGetBoundingClientRect({
    width: vertical ? 0 : 100,
    height: vertical ? 100 : 0,
    top: vertical ? 0 : 100,
    left: vertical ? 100 : 0,
  });
}

describe('Slider', () => {
  test('drag button', async () => {
    const restoreMock = mockRect();
    const wrapper = mount(Slider, {
      propsData: {
        value: 50,
        disabled: true,
      },
      listeners: {
        change(value) {
          wrapper.setProps({ value });
        },
      },
    });
    const button = wrapper.find('.byted-slider__button');
    triggerDrag(button, 50, 0);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.emitted('drag-start')).toBeFalsy();
    expect(wrapper.emitted('drag-end')).toBeFalsy();

    await wrapper.setData({ disabled: false });
    await trigger(button, 'touchstart', 0, 0);
    await trigger(button, 'touchend', 0, 0);
    await triggerDrag(button, 50, 0);
    await Vue.nextTick();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.emitted('drag-start')).toBeTruthy();
    expect(wrapper.emitted('drag-end')).toBeTruthy();

    restoreMock();
  });
});
