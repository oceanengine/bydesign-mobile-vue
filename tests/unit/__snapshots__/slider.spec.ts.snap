// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider drag button 1`] = `
<div class="byted-slider byted-slider--disabled">
  <div class="byted-slider__bar" style="width: 50%;">
    <div class="byted-slider__bar__wrapper">
      <div class="byted-slider__button"></div>
    </div>
  </div>
</div>
`;

exports[`Slider drag button 2`] = `
<div class="byted-slider">
  <div class="byted-slider__bar" style="width: 100%;">
    <div class="byted-slider__bar__wrapper">
      <div class="byted-slider__button"></div>
    </div>
  </div>
</div>
`;
